renderlib/AreaLightTool.cpp

```diff
@@ -0,0 +1,59 @@
+#include "AreaLightTool.h"
+
+void
+AreaLightTool::action(SceneView& scene, Gesture& gesture)
+{
+}
+void
+AreaLightTool::draw(SceneView& scene, Gesture& gesture)
+{
+  if (!scene.scene) {
+    return;
+  }
+  const Light& l = scene.scene->m_lighting.m_Lights[1];
+  glm::vec3 p = l.m_P;
+  glm::vec3 t = l.m_Target;
+  float scale = l.m_Width * 0.5;
+  // compute 4 vertices of square area light pointing at 0, 0, 0
+  glm::vec3 v0 = l.m_U * (-scale) + l.m_V * (-scale);
+  glm::vec3 v1 = l.m_U * scale + l.m_V * (-scale);
+  glm::vec3 v2 = l.m_U * scale + l.m_V * scale;
+  glm::vec3 v3 = l.m_U * (-scale) + l.m_V * scale;
+
+  glm::vec3 color = glm::vec3(1, 1, 1);
+  float opacity = 1.0f;
+  uint32_t code = Gesture::Graphics::SelectionBuffer::k_noSelectionCode;
+  gesture.graphics.addCommand(GL_LINES);
+  gesture.graphics.addLine(Gesture::Graphics::VertsCode(p + v0, color, opacity, code),
+                           Gesture::Graphics::VertsCode(p + v1, color, opacity, code));
+  gesture.graphics.extLine(Gesture::Graphics::VertsCode(p + v2, color, opacity, code));
+  gesture.graphics.extLine(Gesture::Graphics::VertsCode(p + v3, color, opacity, code));
+  gesture.graphics.extLine(Gesture::Graphics::VertsCode(p + v0, color, opacity, code));
+
+  // add translucent quads too?
+  opacity = 0.3f;
+  gesture.graphics.addCommand(GL_TRIANGLES);
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v0, color, opacity, code));
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v1, color, opacity, code));
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v2, color, opacity, code));
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v0, color, opacity, code));
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v2, color, opacity, code));
+  gesture.graphics.addVert(Gesture::Graphics::VertsCode(p + v3, color, opacity, code));
+
+  gesture.graphics.addCommand(GL_LINES);
+  gesture.graphics.addLine(Gesture::Graphics::VertsCode(p + v0 * 0.1f, color, opacity, code),
+                           Gesture::Graphics::VertsCode(t + v0 * 0.1f, color, opacity, code));
+  gesture.graphics.addLine(Gesture::Graphics::VertsCode(p + v1 * 0.1f, color, opacity, code),
+                           Gesture::Graphics::VertsCode(t + v1 * 0.1f, color, opacity, code));
+  gesture.graphics.addLine(Gesture::Graphics::VertsCode(p + v2 * 0.1f, color, opacity, code),
+                           Gesture::Graphics::VertsCode(t + v2 * 0.1f, color, opacity, code));
+  gesture.graphics.addLine(Gesture::Graphics::VertsCode(p + v3 * 0.1f, color, opacity, code),
+                           Gesture::Graphics::VertsCode(t + v3 * 0.1f, color, opacity, code));
```

Comment:
This code is drawing lines from the light towards the target, right? Is there a reason for it not to be grouped with the existing line drawings above? (redundant call to `gesture.graphics.addCommand(GL_LINES)`?)

Also might be nice to add a comment for each of these blocks to say what they're drawing